src/modules/navigator/rtl.cpp

```diff
@@ -688,15 +687,16 @@ float RTL::calculate_return_alt_from_cone_half_angle(float cone_half_angle_deg)
 
 	float return_altitude_amsl = _destination.alt + _param_rtl_return_alt.get();
 
+	// If we are close enough, arbitrarily set altitude proportionally to distance to the destination
 	if (destination_dist <= _navigator->get_acceptance_radius()) {
 		return_altitude_amsl = _destination.alt + 2.0f * destination_dist;
 
 	} else {
-
-		if (cone_half_angle_deg > 0.0f && destination_dist <= _param_rtl_min_dist.get()) {
+		// Only apply cone logic if the half cone angle is set by the user
+		if (!matrix::isEqualF(cone_half_angle_deg, 0.0f) && destination_dist <= _param_rtl_min_dist.get()) {
 
 			// constrain cone half angle to meaningful values. All other cases are already handled above.
-			const float cone_half_angle_rad = radians(constrain(cone_half_angle_deg, 1.0f, 89.0f));
+			const float cone_half_angle_rad = radians(constrain((float)cone_half_angle_deg, 1.0f, 89.0f));
```

Comment:
This C-style cast shouldn't be required here, should it?  It is passed in as a float argument:
`float RTL::calculate_return_alt_from_cone_half_angle(const float cone_half_angle_deg)`

Reply:
Yep you are right :+1: